beacon-chain/monitor/process_sync_committee.go

```diff
@@ -0,0 +1,75 @@
+package monitor
+
+import (
+	"fmt"
+
+	"github.com/prysmaticlabs/prysm/beacon-chain/state"
+	ethpb "github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/proto/prysm/v1alpha1/block"
+	"github.com/sirupsen/logrus"
+)
+
+// processSyncCommitteeContribution logs the event that one of our tracked
+// validators' aggregated sync contribution has been processed.
+// TODO: We do not log if a sync contribution was included in an aggregate (we
+// log them when they are included in blocks)
+func (s *Service) processSyncCommitteeContribution(contribution *ethpb.SignedContributionAndProof) {
+	idx := contribution.Message.AggregatorIndex
+	if s.TrackedIndex(idx) {
+		aggPerf := s.aggregatedPerformance[idx]
+		aggPerf.totalSyncComitteeAggregations++
+		s.aggregatedPerformance[idx] = aggPerf
+
+		log.WithField("ValidatorIndex", contribution.Message.AggregatorIndex).Info("Sync committee aggregation processed")
+	}
+}
+
+// processSyncAggregate logs the event that one of our tracked validators is a sync-committee member and its
+// contribution was included
+func (s *Service) processSyncAggregate(state state.BeaconState, blk block.BeaconBlock) {
+	if blk == nil || blk.Body() == nil {
+		return
+	}
+	bits, err := blk.Body().SyncAggregate()
+	if err != nil {
+		log.WithError(err).Error("Cannot get SyncAggregate")
+		return
+	}
+	for validatorIdx, committeeIndices := range s.trackedSyncCommitteeIndices {
+		if len(committeeIndices) > 0 {
+			contrib := 0
+			for _, idx := range committeeIndices {
+				if bits.SyncCommitteeBits.BitAt(uint64(idx)) {
+					contrib++
+				}
+			}
+
+			balance, err := state.BalanceAtIndex(validatorIdx)
+			if err != nil {
+				log.Error("Could not get balance")
+				return
+			}
+
+			latestPerf := s.latestPerformance[validatorIdx]
+			balanceChg := balance - latestPerf.balance
+			latestPerf.balanceChange = balanceChg
+			latestPerf.balance = balance
+			s.latestPerformance[validatorIdx] = latestPerf
```

Comment:
This seems like a shared resource. Could this map ever be updated concurrently? If so we would need a lock

Reply:
I was going to study sync map, but decided to go with Locks everywhere, let me know what you think, there's an unexported helper function that assumes a lock inside.